test/mocha/fixturesUI.js

```diff
@@ -0,0 +1,134 @@
+const commonSuite = require('mocha/lib/interfaces/common');
+
+Error.stackTraceLimit = 15;
+global.testOptions = require('../harness/testOptions');
+global.registerFixture = registerFixture;
+global.registerWorkerFixture = registerWorkerFixture;
+process.env.JEST_WORKER_ID = 1;
+
+const fixturePool = new FixturePool();
+let revertBabelRequire;
+
+function fixturesUI(suite) {
+  const suites = [suite];
+
+  suite.on(Suite.constants.EVENT_FILE_PRE_REQUIRE, function(context, file, mocha) {
+    const common = commonSuite(suites, context, mocha);
+
+    context.beforeEach = common.beforeEach;
+    context.afterEach = common.afterEach;
+    fixturePool.patchToEnableFixtures(context, 'beforeEach');
+    fixturePool.patchToEnableFixtures(context, 'afterEach');
+
+    context.run = mocha.options.delay && common.runWithSuite(suite);
+
+    context.describe = (title, fn) => {
+      return common.suite.create({
+        title: title,
+        file: file,
+        fn: fn
+      });
+    };
+
+    context.xdescribe = (title, fn) => {
+      return common.suite.skip({
+        title: title,
+        file: file,
+        fn: fn
+      });
+    };
+
+    context.describe.skip = function(condition) {
+      return condition ? context.xdescribe : context.describe;
+    };
+
+    context.describe.only = (title, fn) => {
+      return common.suite.only({
+        title: title,
+        file: file,
+        fn: fn
+      });
+    };
+
+    context.fdescribe = context.describe.only;
+
+    context.it = context.specify = function(title, fn) {
+      const suite = suites[0];
+      if (suite.isPending()) {
+        fn = null;
+      }
+      const wrapped = fixturePool.wrapTestCallback(fn);
+      const wrapper = wrapped ? (done, ...args) => {
+        wrapped(...args).then(done).catch(done);
+      } : undefined;
+
+      const test = new Test(title, wrapper);
+      test.file = file;
+      suite.addTest(test);
+      return test;
+    };
+
+    context.it.only = function(title, fn) {
+      return common.test.only(mocha, context.it(title, fn));
+    };
+
+    context.fit = context.it.only;
+
+    context.xit = function(title) {
+      return context.it(title);
+    };
+
+    context.it.skip = function(condition) {
+      return condition ? context.xit : context.it;
+    };
+
+    context.it.fail = function(condition) {
+      return condition ? context.xit : context.it;
+    };
+
+    context.it.slow = function(condition) {
+      return context.it;
+    };
+
+    context.it.retries = function(n) {
+      context.retries(n);
+    };
+
+    revertBabelRequire = pirates.addHook((code, filename) => {
+      const result = babel.transformFileSync(filename, {
+        presets: [
+          ['@babel/preset-env', {targets: {node: 'current'}}],
+          '@babel/preset-typescript']
+      });
+      return result.code;
+    }, {
+      exts: ['.ts']
+    });
+  });
+
+  suite.on(Suite.constants.EVENT_FILE_POST_REQUIRE, function(context, file, mocha) {
```

Comment:
This will fail if we do any dynamic requires right? I think we are safe because its node_modules, but we do that in `base.fixtures.js` to require playwright after we've set it to test mode. It also seems like an easy pitfall.

Reply:
I am not sure I follow your concern. Let's not do base.fixtures.js if that is a problem?